samples/apply-colormap-renderer-to-raster/src/main/java/com/esri/arcgismaps/sample/applycolormaprenderertoraster/components/ApplyColormapRendererToRasterViewModel.kt

```diff
@@ -0,0 +1,87 @@
+/* Copyright 2025 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.applycolormaprenderertoraster.components
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.Color
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.RasterLayer
+import com.arcgismaps.mapping.symbology.raster.ColormapRenderer
+import com.arcgismaps.raster.Raster
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.launch
+import java.io.File
+
+/**
+ * ViewModel that creates a map with a raster layer and applies a colormap renderer to the raster.
+ *
+ * The sample expects the raster file to be placed in the application's external files directory
+ * under a folder matching the sample name and a subfolder "raster-file" containing "Shasta.tif".
+ * This mirrors the behavior of other samples that provision offline assets. If the raster is not
+ * present the ViewModel will attempt to load and will surface an error through [messageDialogVM].
```

Comment:
Could be removed, the top comment seems descriptive enough: 


```suggestion
 * The sample expects the raster file to be placed in the application's external files directory
 * under a folder matching the sample name and a subfolder "raster-file" containing "Shasta.tif".
```